base/c.jl

```diff
@@ -43,6 +43,29 @@ end
 
 typealias Coff_t FileOffset
 
+# C NUL-terminated string pointers; these can be used in ccall
+# instead of Ptr{UInt8} and Ptr{Cwchar_t}, respectively, to enforce
+# a check for embedded NUL chars in the string (to avoid silent truncation).
+immutable Cstring; p::Ptr{UInt8}; end
```

Comment:
Should this be `Cchar` although I'm not familiar with any systems where `Cchar != UInt8` and we're likely to be broken for many reasons on any such system?